pkgs/development/compilers/tinygo/default.nix

```diff
@@ -39,36 +38,37 @@ in
 
 buildGoModule rec {
   pname = "tinygo";
-  version = "0.26.0";
+  version = "0.30.0";
 
   src = fetchFromGitHub {
     owner = "tinygo-org";
     repo = "tinygo";
     rev = "v${version}";
-    sha256 = "rI8CADPWKdNvfknEsrpp2pCeZobf9fAp0GDIWjupzZA=";
+    sha256 = "sha256-hOccfMKuvTKYKDRcEgTJ8k/c/H+qNDpvotWIqk6p2u8=";
     fetchSubmodules = true;
   };
 
-  vendorHash = "sha256-ihQd/RAjAQhgQZHbNiWmAD0eOo1MvqAR/OwIOUWtdAM=";
+  vendorHash = "sha256-2q3N6QhfRmwbs4CTWrFWr1wyhf2jPS2ECAn/wrrpXdM=";
 
   patches = [
     ./0001-Makefile.patch
 
     (substituteAll {
       src = ./0002-Add-clang-header-path.patch;
-      clang_include = "${clang.cc.lib}/lib/clang/${clang.cc.version}/include";
+      clang_include = "${clang.cc.lib}/lib/clang/${llvmMajor}/include";
     })
 
     #TODO(muscaln): Find a better way to fix build ID on darwin
     ./0003-Use-out-path-as-build-id-on-darwin.patch
   ];
 
-  nativeCheckInputs = [ avrgcc binaryen ];
+  nativeCheckInputs = [ binaryen ];
   nativeBuildInputs = [ makeWrapper ];
   buildInputs = [ llvm clang.cc ]
     ++ lib.optionals stdenv.isDarwin [ zlib ncurses libffi libxml2 xar ];
 
-  doCheck = (stdenv.buildPlatform.canExecute stdenv.hostPlatform);
+  # can't get tests to run on darwin: fatal error: 'fcntl.h' file not found
+  doCheck = (stdenv.buildPlatform.canExecute stdenv.hostPlatform) && stdenv.isLinux;
```

Comment:
This can be fixed by the following small patch to TinyGo:

```diff
--- a/compileopts/config.go
+++ b/compileopts/config.go
@@ -269,6 +269,7 @@ func (c *Config) CFlags() []string {
                root := goenv.Get("TINYGOROOT")
                cflags = append(cflags,
                        "--sysroot="+filepath.Join(root, "lib/macos-minimal-sdk/src"),
+                       "-isystem", filepath.Join(root, "lib/macos-minimal-sdk/src/usr/include"), // necessary for Nix
                )
        case "picolibc":
                root := goenv.Get("TINYGOROOT")
```

Not sure why Nix Clang doesn't seem to use `--sysroot` include directories, but this is a workaround.

Reply:
added patch